src/requests/cache/sorted_set_fetch_by_rank.rs

```diff
@@ -0,0 +1,143 @@
+use momento_protos::cache_client::sorted_set_fetch_request::{by_index, ByIndex, Range};
+use momento_protos::cache_client::{SortedSetFetchRequest, Unbounded};
+
+use crate::requests::cache::sorted_set_fetch_by_rank::SortOrder::Ascending;
+use crate::requests::cache::MomentoRequest;
+use crate::response::cache::sorted_set_fetch::SortedSetFetch;
+use crate::simple_cache_client::prep_request_with_timeout;
+use crate::{CacheClient, IntoBytes, MomentoResult};
+
+#[repr(i32)]
+pub enum SortOrder {
+    Ascending = 0,
+    Descending = 1,
+}
+
+/// Request to fetch the elements in a sorted set by their rank.
+///
+/// # Arguments
+///
+/// * `cache_name` - The name of the cache containing the sorted set.
+/// * `sorted_set_name` - The name of the sorted set to add an element to.
+///
+/// # Optional Arguments
+///
+/// * `order` - The order to sort the elements by. [SortOrder::Ascending] or [SortOrder::Descending].
+/// Defaults to Ascending.
+/// * `start_rank` - The rank of the first element to fetch. Defaults to 0. This rank is
+/// inclusive, i.e. the element at this rank will be fetched.
+/// * `end_rank` - The rank of the last element to fetch. This rank is exclusive, i.e. the
+/// element at this rank will not be fetched. Defaults to -1, which fetches up until and
+/// including the last element.
+///
+/// # Examples
+/// Assumes that a CacheClient named `cache_client` has been created and is available.
+/// ```
+/// # fn main() -> anyhow::Result<()> {
+/// # use std::convert::TryInto;
+/// # use momento_test_util::create_doctest_client;
+/// # tokio_test::block_on(async {
+/// use momento::requests::cache::sorted_set_fetch_by_rank::SortOrder;
+/// use momento::requests::cache::sorted_set_fetch_by_rank::SortedSetFetchByRankRequest;
+/// use momento::response::cache::sorted_set_fetch::SortedSetFetch;
+/// # let (cache_client, cache_name) = create_doctest_client();
```

Comment:
do you know if this will get excluded from the generated docs, or whether the comment blocks have to be contiguous at the top/bottom of the function in order to get excluded? if you don't know we can just look into this as a follow-up, not a blocker for this PR.

Reply:
It will exclude any line maked with a hash. Here is what that example looks like:
```
use momento::requests::cache::sorted_set_fetch_by_rank::SortOrder;
use momento::requests::cache::sorted_set_fetch_by_rank::SortedSetFetchByRankRequest;
use momento::response::cache::sorted_set_fetch::SortedSetFetch;
let sorted_set_name = "sorted_set";

let put_element_response = cache_client.sorted_set_put_elements(
    cache_name.to_string(),
    sorted_set_name.to_string(),
    vec![("value1", 1.0), ("value2", 2.0), ("value3", 3.0), ("value4", 4.0)]
).await?;
```

Reply:
sweet, thanks